### PR TITLE
Fix merge stream race condition

### PR DESCRIPTION
**Pull request by sveisvei**
cc @trygve-lie @runeh @SimenB

### CODE REVIEW

File: lib/reader.js [hunk 1]
```diff
@@ -61,12 +61,18 @@ module.exports = class Reader extends pack {
...
         const merged = mergeStream();
         let count = 0;
-        streams.forEach((strm, index) => {
-            const orderDecorator = new OrderDecorator(index);
+        streams.forEach((readStream, index) => {
+            const tmpStream = new stream.PassThrough({ objectMode: true });
+            merged.add(tmpStream);
+
+            readStream.on('file found', (file) => {
+                const orderDecorator = new OrderDecorator(index);
```
Comment (SimenB) [inline lib/reader.js]: why inside of the callback?

Comment (sveisvei) [inline lib/reader.js]: Only create if `file found` makes sense to me.

**Comment by trygve-lie** [comment]
lgtm

File: test/reader.test.js [hunk 2]
```diff
@@ -78,13 +82,104 @@ test('should keep ordering even if slow', async (t) => {
...
+        return myStream;
+    }
+
+    const sink = new FileSystemSink({ path: path.join(__dirname, 'mock') });
+    const feedA = createStream(sink, 'simple.a.json', 200);
+    const feedB = sink.reader('simple.b.json');
+    const feedC2 = createStream(sink, 'simple.c.json', 200);
+
+    sink.on('file saved', (id, file) => {
+        console.log(id, file);
```
Comment (SimenB) [inline test/reader.test.js]: do we want this?

Comment (sveisvei) [inline test/reader.test.js]: Lol no. 🤔

**Pull request merged**